apis/v1beta1/targetallocator_types.go

```diff
@@ -15,125 +15,28 @@
-	// +optional
-	// +listType=atomic
-	// +kubebuilder:pruning:PreserveUnknownFields
-	ScrapeConfigs []AnyConfig `json:"scrapeConfigs,omitempty"`
-	// PrometheusCR defines the configuration for the retrieval of PrometheusOperator CRDs ( servicemonitor.monitoring.coreos.com/v1 and podmonitor.monitoring.coreos.com/v1 ).
-	// +optional
-	PrometheusCR TargetAllocatorPrometheusCR `json:"prometheusCR,omitempty"`
-	// ObservabilitySpec defines how telemetry data gets handled.
-	//
-	// +optional
-	// +kubebuilder:validation:Optional
-	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Observability"
-	Observability ObservabilitySpec `json:"observability,omitempty"`
-}
-
-// TargetAllocatorPrometheusCR configures Prometheus CustomResource handling in the Target Allocator.
-type TargetAllocatorPrometheusCR struct {
-	// Enabled indicates whether to use a PrometheusOperator custom resources as targets or not.
-	// +optional
-	Enabled bool `json:"enabled,omitempty"`
-	// Default interval between consecutive scrapes. Intervals set in ServiceMonitors and PodMonitors override it.
-	//Equivalent to the same setting on the Prometheus CR.
-	//
-	// Default: "30s"
-	// +kubebuilder:default:="30s"
-	// +kubebuilder:validation:Format:=duration
-	ScrapeInterval *metav1.Duration `json:"scrapeInterval,omitempty"`
-	// PodMonitors to be selected for target discovery.
-	// This is a map of {key,value} pairs. Each {key,value} in the map is going to exactly match a label in a
-	// PodMonitor's meta labels. The requirements are ANDed.
-	// +optional
-	PodMonitorSelector *metav1.LabelSelector `json:"podMonitorSelector,omitempty"`
-	// ServiceMonitors to be selected for target discovery.
-	// This is a map of {key,value} pairs. Each {key,value} in the map is going to exactly match a label in a
-	// ServiceMonitor's meta labels. The requirements are ANDed.
-	// +optional
-	ServiceMonitorSelector *metav1.LabelSelector `json:"serviceMonitorSelector,omitempty"`
-}
-
 type (
-	// TargetAllocatorAllocationStrategy represent a strategy Target Allocator uses to distribute targets to each collector
-	// +kubebuilder:validation:Enum=least-weighted;consistent-hashing;per-node
-	TargetAllocatorAllocationStrategy string
-	// TargetAllocatorFilterStrategy represent a filtering strategy for targets before they are assigned to collectors
-	// +kubebuilder:validation:Enum="";relabel-config
-	TargetAllocatorFilterStrategy string
+	// TargetAllocatorPrometheusCR configures Prometheus CustomResource handling in the Target Allocator.
+	TargetAllocatorPrometheusCR common.TargetAllocatorPrometheusCR
+	// TargetAllocatorAllocationStrategy represent a strategy Target Allocator uses to distribute targets to each collector.
+	TargetAllocatorAllocationStrategy common.TargetAllocatorAllocationStrategy
+	// TargetAllocatorFilterStrategy represent a filtering strategy for targets before they are assigned to collectors.
+	TargetAllocatorFilterStrategy common.TargetAllocatorFilterStrategy
```

Comment:
should this remain here? Or is this because we need to maintain a v1alpha1 and v1beta1 for this now?